src/main/java/org/opensearch/security/securityconf/ConfigModelV7.java

```diff
@@ -366,19 +366,29 @@ public EvaluatedDlsFlsConfig getDlsFls(User user, IndexNameExpressionResolver re
             Map<String, Set<String>> dlsQueriesByIndex = new HashMap<String, Set<String>>();            
             Map<String, Set<String>> flsFields = new HashMap<String, Set<String>>();
             Map<String, Set<String>> maskedFieldsMap = new HashMap<String, Set<String>>();
+
+            // we capture all concrete indices that do not have any
+            // DLS/FLS/Masked Fields restrictions. If the dfm_empty_overwrites_all
+            // switch is enabled, this trumps any restrictions on those indices
+            // that may be imposed by other roles.
+            Set<String> noDlsConcreteIndices = new HashSet<>();
+            Set<String> noFlsConcreteIndices = new HashSet<>();
+            Set<String> noMaskedFieldConcreteIndices = new HashSet<>();
                         
             for (SecurityRole role : roles) {
                 for (IndexPattern ip : role.getIpatterns()) {
 					Set<String> concreteIndices;
-					concreteIndices = ip.getResolvedIndexPattern(user, resolver, cs);
+					concreteIndices = ip.getResolvedIndexPattern(user, resolver, cs, false);
 					String dls = ip.getDlsQuery(user);
 
 					if (dls != null && dls.length() > 0) {
 
 						for (String concreteIndex : concreteIndices) {
 							dlsQueriesByIndex.computeIfAbsent(concreteIndex, (key) -> new HashSet<String>()).add(dls);
 						}
-					}
+					} else if (dfmEmptyOverwritesAll) {
+                        noDlsConcreteIndices.addAll(concreteIndices);
```

Comment:
the formatting is still off here and should be fixed (even if no linter is in place for this yet)

Reply:
The indentation looks good in my IDE. This looks like space vs tab issue in Github. Similar issue like [this](https://github.com/github/VisualStudio/issues/1524#issuecomment-372632866).
<img width="686" alt="Screen Shot 2022-04-09 at 10 02 42 PM" src="https://user-images.githubusercontent.com/16925774/162602225-2a983ea2-0b58-4bd3-830e-78f0652bf114.png">

Reply:
The indentation looks good in my IDE. This looks like space vs tab issue in Github. Similar issue like https://github.com/github/VisualStudio/issues/1524#issuecomment-372632866.
<img width="634" alt="Screen Shot 2022-04-09 at 10 55 46 PM" src="https://user-images.githubusercontent.com/16925774/162604435-3f0e66f6-903f-4fd8-a91a-0f86e0dc6ca9.png">

Reply:
Fixed.